src/projects/digital/site/src/containers/DigitalMainDesigner/index.tsx

```diff
@@ -8,11 +8,70 @@ import {MainDesigner} from "shared/site/containers/MainDesigner";
 import {useMainDigitalDesigner} from "digital/site/utils/hooks/useDigitalDesigner";
 
 import "./index.scss";
+import {DigitalCircuit} from "digital/api/circuit/public";
+import {SmartPlaceOptions} from "digital/site/utils/DigitalCreate";
+import {V, Vector} from "Vector";
+import {Rect} from "math/Rect";
 
 // import {useMainCircuit} from "shared/site/utils/hooks/useCircuit";
 
 
-export const DigitalMainDesigner = () => {
+const AUTO_PLACE_SWITCH_SPACE = 2.5;
+const AUTO_PLACE_LED_SPACE = 1;
+/**
+ * Utility function that, given a DigitalComponent id, will create the component N times vertically
+ *  (with behavior matches DigitalCreateN) but also create Switches for each input and LEDs for each
+ *  output and automatically connect them together. Starts placing at position `pos`.
+ * This function is directly used for implementation of issue #689.
+ *
+ * @param pos     The position of the first component.
+ * @param itemId  The ID of the item, if an IC then it has the form: `ic/INDEX`, where INDEX
+ *                corresponds to the index of the IC relative to the list of ICs in `designer`.
+ * @param circuit The cirucit the items. Needed to place items.
+ * @param N       The number of items to create.
+ * @param options The options used to indicate what connected components to create.
+ * @throws If the itemId is an invalid item or IC.
+ */
+export function SmartPlace(pos: Vector, itemId: string, circuit: DigitalCircuit,
+    N: number, options: SmartPlaceOptions) {
+    circuit.beginTransaction();
+    for (let i = 0; i < N; i++) {
+        const comp = circuit.placeComponentAt(itemId, pos);
+
+        // Need to do it like this rather then comp.getInputPorts() since this can
```

Comment:
This is technically still accurate, `comp.inputs` would return the group labeled "inputs", not all the input ports (at the moment) -- maybe it would make more sense for it to return all the ports that are input ports but im not sure if that would be confusing or not, wdyt? This would also make a clear decision on the "undefined vs []" problem